src/Doctrine/BaseRelation.php

```diff
@@ -19,8 +19,11 @@ abstract class BaseRelation
     private $propertyName;
     private $targetClassName;
     private $targetPropertyName;
+    private $returnType;
+    private $returnTypeIsNullable = true;
```

Comment:
```suggestion
    private $returnTypeIsNullable;
```

I just don't want this to have a default value... so let's leave it null.

Reply:
And maybe we should call these properties:`overriddenReturnType` and `isOverriddenReturnTypeNullable`... because that's what they *really* are. If they are not set, then it means that the "default" return type of the related class would be used.